web/web/auth/static/actions.js

```diff
@@ -1,6 +1,99 @@
-export function loginSuccessful(token) {
+import Cookies from "js-cookie";
+import { api } from "../../static/js/network_client";
+import { createLoginError } from "./helpers";
+
+export function loginSuccessful() {
     return {
         type: "LOGIN_SUCCESSFUL",
-        token
-    }
-}
\ No newline at end of file
+        userLoggedIn: true,
+    };
+}
+
+export function loginFailed() {
+    return {
+        type: "LOGIN_FAILED",
+        userLoggedIn: false,
+    };
+}
+
+export function resetUserLoggedIn() {
+    return {
+        type: "RESET_USER_LOGGED_IN",
+        userLoggedIn: false,
+    };
+}
+
+export function processLogin(username, password) {
+    return (dispatch) => {
+        try {
+            const json = {
```

Comment:
can we call this `payload` instead? having this be `json["json"]` seems weird. And why does it need to be nested?

Reply:
Yes can change to payload. From what I read on Ky, it requires the data  (headers, body, etc) to be nested, didn't work for me un-nested. https://github.com/sindresorhus/ky

Reply:
if you use:
```javascript
const json = { username: .... }
...
api.post("login", {...json}) 
```
maybe even `json` w/o the spread would work...